test/test_start_date.py

```diff
@@ -90,32 +90,22 @@ def run_test(self, days, expected_streams):
                 # collect information for assertions from syncs 1 & 2 base on expected values
                 record_count_sync_1 = record_count_by_stream_1.get(stream, 0)
                 record_count_sync_2 = record_count_by_stream_2.get(stream, 0)
-                primary_keys_list_1 = [tuple(message.get('data').get(expected_pk)
-                                             for expected_pk in expected_primary_keys)
-                                       for message in synced_records_1.get(stream, {}).get('messages', [])
-                                       if message.get('action') == 'upsert']
+                primary_keys_list_1 =  [tuple(message.get('data').get('pk') for pk in expected_primary_keys)
```

Comment:
I think this is the bug (single quotes around the pk in get('pk').
```suggestion
                primary_keys_list_1 = [tuple(message.get('data').get(pk) for pk in expected_primary_keys)
```